apps/game_backend/lib/game_backend/items/item_cost.ex

```diff
@@ -0,0 +1,26 @@
+defmodule GameBackend.Items.ItemCost do
+  @moduledoc """
+  Contains a set the different prices to purchase an ItemTemplate.
+  """
+  alias GameBackend.Users.Currencies.CurrencyCost
+  alias GameBackend.Items.ItemTemplate
+
+  use GameBackend.Schema
+  import Ecto.Changeset
+
+  schema "item_costs" do
+    field(:name, :string)
+    belongs_to(:item_template, ItemTemplate)
+    embeds_many(:currency_costs, CurrencyCost)
```

Comment:
Why are we creating a new schema to embed the CurrencyCost instead of just embedding it in the ItemTemplate itself? Also, why does this ItemCost have a name?

Is it to be able to purchase items in two different sets of currencies?

Reply:
If it is, I think it'd be good to leave this documented in the moduledoc as it may not be super intuitive (even though it is a great approach).

I'm not sure I'm a big fan of the "name" string as it is right now. It seems too arbitrary which could be an issue for when we need to develop around it. At the very least, I feel like adding an inclusion constraint so that we know which values this name can take is necessary.

Reply:
I'm pointing my final solution towards the following:
- A store that has many buyables.
- A buyable will have:
  - ItemCosts (different combinations of {currency, amount}.
  - One of (buyable_type):
    - Currency
    - ItemTemplate
    - Etc.
![Screenshot 2024-05-27 at 4 39 53 PM](https://github.com/lambdaclass/mirra_backend/assets/65268603/8266f29b-90c8-47db-a05f-b8069230863e)

> why does this ItemCost have a name?
The client will send us this name to identify which set of currencies we should get. We can talk to achieve a better solution to this.
I'll also improve the moduledoc.

Reply:
I like that implementation a lot! Doesn't it make the item_costs name useless? You have some predefined stores that you inform the client about, and then the client asks for a specific store in order to get all the different Buyables from that store. I see no need for `item_costs.name` now

Reply:
After what we talked at the office w/ @marco-paulucci and the pictures from Jaz (see figure 1), I'll remove item_costs assoc and I'll just make a embeds_many from buyables to currency_costs.
Figure 1:
![Screenshot 2024-05-27 at 5 19 16 PM](https://github.com/lambdaclass/mirra_backend/assets/65268603/a98af5c0-e0c3-4bc1-b835-37a47d9a4a1a)